Library/Homebrew/formulary.rb

```diff
@@ -182,7 +182,8 @@ def self.load_formula_from_api(name, flags:)
       end
 
       if (keg_only_reason = json_formula["keg_only_reason"]).present?
-        reason = Formulary.convert_to_string_or_symbol keg_only_reason["reason"]
+        reason = keg_only_reason["reason"]&.gsub(HOMEBREW_PREFIX_PLACEHOLDER, HOMEBREW_PREFIX)
```

Comment:
I'm wondering if extracting this to `def replace_prefix_placeholder(string)` or something would be nice so this can be shared for `caveats` too?